base/file.jl

```diff
@@ -48,12 +46,22 @@ julia> pwd()
 ```
 """
 function pwd()
-    b = Vector{UInt8}(undef, 1024)
-    len = RefValue{Csize_t}(length(b))
-    uv_error(:getcwd, ccall(:uv_cwd, Cint, (Ptr{UInt8}, Ptr{Csize_t}), b, len))
-    String(b[1:len[]])
+    buf = Base.StringVector(MAX_PATH - 1) # space for null-terminator implied by StringVector
+    sz = RefValue{Csize_t}(length(buf) + 1) # total buffer size including null
+    while true
+        rc = ccall(:uv_cwd, Cint, (Ptr{UInt8}, Ptr{Csize_t}), buf, sz)
+        if rc == 0
+            resize!(buf, sz[])
+            return String(buf)
+        elseif rc == Base.UV_ENOBUFS
+            resize!(buf, sz[] - 1) # space for null-terminator implied by StringVector
+        else
+            uv_error(:cwd, rc)
```

Comment:
should we call this `"pwd"` to reflect the name of the function or `"getcwd"` to reflect the name of the libc function?

Reply:
Good question. I called it `cwd` to reflect the name of the libuv call, which I decided was perhaps more helpful for debugging purposes.   If the `pwd()` function errors in julia it will be clear which function is throwing the error from the stacktrace, but perhaps not clear which internal function caused the error and thus I decided to call it `:cwd` instead of `pwd`.